src/storage/mongodb/ChargingStationStorage.ts

```diff
@@ -707,6 +707,33 @@ export default class ChargingStationStorage {
       });
     }
 
+    if (params.withChargingStation || params.withSiteArea) {
+      // Charging Stations
+      DatabaseUtils.pushChargingStationLookupInAggregation({
+        tenantID, aggregation, localField: 'chargingStationID', foreignField: '_id',
+        asField: 'chargingStation', oneToOneCardinality: true, oneToOneCardinalityNotNull: false
+      });
+      // Site Areas
+      DatabaseUtils.pushSiteAreaLookupInAggregation({
+        tenantID, aggregation, localField: 'chargingStation.siteAreaID', foreignField: '_id',
+        asField: 'chargingStation.siteArea', oneToOneCardinality: true, oneToOneCardinalityNotNull: false
+      });
+      // Convert
+      DatabaseUtils.pushConvertObjectIDToString(aggregation, 'chargingStation.siteAreaID');
```

Comment:
This you can let it at the end.
If you don't get the meaning of these lines your won't know where to put them.
I moved them.